test/performance/odp_bench_pktio_sp.c

```diff
@@ -615,16 +597,204 @@ static int pktout_event_queue_stats(bench_tm_result_t *res, int repeat_count)
+{
+	*term = ODP_PMR_TCP_DPORT;
+
+	if (capa->supported_terms.bit.udp_sport)
+		*term = ODP_PMR_UDP_SPORT;
+	else if (capa->supported_terms.bit.udp_dport)
+		*term = ODP_PMR_UDP_DPORT;
+	else if (capa->supported_terms.bit.tcp_sport)
+		*term = ODP_PMR_TCP_SPORT;
+	else if (capa->supported_terms.bit.tcp_dport)
+		*term = ODP_PMR_TCP_DPORT;
+	else
+		return 0;
+
+	return 1;
+}
+
+/* Capabilities required for cls_create_pmr() test */
+static int check_cls_capa(void)
+{
+	appl_args_t *appl_args = gbl_args;
+	odp_cls_capability_t cls_capa;
+	odp_schedule_capability_t sched_capa;
+	int ret;
+
+	ret = odp_cls_capability(&cls_capa);
+	if (ret) {
+		ODPH_ERR("Reading classifier capa failed: %d\n", ret);
+		return -1;
+	}
+
+	ret = odp_schedule_capability(&sched_capa);
+	if (ret) {
+		ODPH_ERR("Reading scheduler capa failed: %d\n", ret);
+		return -1;
+	}
+
+	if (!find_first_supported_l3_pmr(&cls_capa, &appl_args->cls_create_pmr.term)) {
+		ODPH_ERR("Implementations doesn't support any TCP/UDP PMRs\n");
+		return 0;
+	}
+
+	/* One extra CoS and queue required for the default CoS */
+	if (appl_args->opt.num_pmr + 1 > cls_capa.max_cos) {
+		ODPH_ERR("Not enough CoSes supported for PMR test: %u/%u\n",
+			 appl_args->opt.num_pmr + 1, cls_capa.max_cos);
+		return 0;
+	}
+
+	if (appl_args->opt.num_pmr + 1 > sched_capa.max_queues) {
+		ODPH_ERR("Not enough queues supported for PMR test: %u/%u\n",
+			 appl_args->opt.num_pmr + 1, sched_capa.max_queues);
+		return 0;
+	}
+
+	appl_args->cls_create_pmr.enabled = true;
+
+	return 1;
+}
+
+static int check_cls_cond(void)
+{
+	return gbl_args->cls_create_pmr.enabled;
+}
+
+static int cls_create_pmr(bench_tm_result_t *res, int repeat_count)
+{
+	appl_args_t *appl_args = gbl_args;
+	odp_pktio_t pktio = appl_args->pktio;
+	odp_cls_cos_param_t cos_param;
+	odp_queue_param_t queue_param;
+	odp_pmr_param_t pmr_param;
+	odp_cos_t default_cos;
+	uint32_t num_cos = appl_args->opt.num_pmr + 1;
+	uint32_t num_pmr = num_cos - 1;
+	uint32_t cos_created = 0;
+	uint32_t queue_created = 0;
+	uint16_t val = 1024;
+	uint16_t mask = 0xffff;
+	int ret = 0;
+	odp_time_t t1, t2;
+	odp_cos_t cos[num_cos];
+	odp_queue_t queue[num_cos];
+	odp_pmr_t pmr[num_pmr];
+	uint8_t id1 = bench_tm_func_register(res, "odp_queue_create()");
+	uint8_t id2 = bench_tm_func_register(res, "odp_cls_cos_create()");
+	uint8_t id3 = bench_tm_func_register(res, "odp_cls_pmr_create()");
+	uint8_t id4 = bench_tm_func_register(res, "odp_cls_pmr_destroy()");
+
+	odp_queue_param_init(&queue_param);
+	queue_param.type = ODP_QUEUE_TYPE_SCHED;
+
+	odp_cls_cos_param_init(&cos_param);
+
+	for (uint32_t i = 0; i < num_cos; i++) {
+		t1 = odp_time_local_strict();
+		queue[i] = odp_queue_create(NULL, &queue_param);
+
+		if (queue[i] == ODP_QUEUE_INVALID) {
+			ODPH_ERR("odp_queue_create() failed %u / %u\n", i + 1, num_cos);
+			break;
+		}
+		t2 = odp_time_local_strict();
+		bench_tm_func_record(t2, t1, res, id1);
```

Comment:
Since queue / CoS destroys are not recorded. Maybe queue/Cos create/destroy could be another test case (added in another PR). So, these create records could be removed from this test case to make it simpler.

Reply:
Changed in v2.